### PR TITLE
feat: error decoding AE

### PR DESCRIPTION
**Pull request by ioanaroceanu2000**

### CODE REVIEW

File: reya-rust-sdk/src/http_provider.rs [hunk 1]
```diff
@@ -606,6 +633,150 @@ fn decode_reason(reason_bytes: Bytes) -> (String, ReasonError) {
...
+                                e
+                            )
+                        }
+                    }
+                }
+                None => info!("[Error decoding] Failed to get error response"),
+            }
+        }
+        _ => {
+            info!("[DEcode] Transaction failed: {:?}", e);
```
Comment (moboware) [inline reya-rust-sdk/src/http_provider.rs:775]: Decode

File: reya-rust-sdk/src/http_provider.rs [hunk 2]
```diff
@@ -429,11 +451,16 @@ impl HttpProvider {
...
-        trace!("End Trigger Auto-exchange");
-
-        eyre::Ok(transaction_result.tx_hash().clone())
+        match builder.send().await {
+            Ok(transaction_result) => {
+                trace!("End Trigger Auto-exchange");
+                return eyre::Ok(transaction_result.tx_hash().clone());
+            }
+            Err(e) => {
+                handle_auto_exchange_error(e);
```
Comment (moboware) [inline reya-rust-sdk/src/http_provider.rs:460]: - handle_auto_Exchange_error looks like a general rpc error handling fn, correct? Is that  case can we make it general usable?
- The errors are only logged, would be better to pass them on to the next layer with the Err(...). Add the error string into the format so we can log it e.g. in data dog logger

File: reya-rust-sdk/src/http_provider.rs [hunk 3]
```diff
@@ -606,6 +633,150 @@ fn decode_reason(reason_bytes: Bytes) -> (String, ReasonError) {
...
+        Err(err) => {
+            return (format!("Error={:?}", err), AEReasonError::DecodingError);
+        }
+    }
+}
+
+fn handle_auto_exchange_error(e: Error) {
+    match e {
+        Error::AbiError(revert_reason) => {
+            info!("[Error decoding] ABI error: {:?}", revert_reason);
```
Comment (moboware) [inline reya-rust-sdk/src/http_provider.rs:744]: errors should be logged as error! in all cases

Comment (ioanaroceanu2000) [reply reya-rust-sdk/src/http_provider.rs]: true, changed all info from this fn into error in [2d88303](https://github.com/Reya-Labs/reya-rust-sdks-monorepo/pull/24/commits/2d883034e2956bb6b0fcdd9eace3cbab6af9b46c)

File: reya-rust-sdk/src/http_provider.rs (hunk 2 above)
Comment (ioanaroceanu2000) [reply reya-rust-sdk/src/http_provider.rs]: I just renamed the fn into `handle_rpc_error`. Also sending the error string upstream if decoded. see [2d88303](https://github.com/Reya-Labs/reya-rust-sdks-monorepo/pull/24/commits/2d883034e2956bb6b0fcdd9eace3cbab6af9b46c)

File: reya-rust-sdk/src/http_provider.rs (hunk 1 above)
Comment (ioanaroceanu2000) [reply reya-rust-sdk/src/http_provider.rs]: fixed [2d88303](https://github.com/Reya-Labs/reya-rust-sdks-monorepo/pull/24/commits/2d883034e2956bb6b0fcdd9eace3cbab6af9b46c)

File: reya-rust-sdk/src/http_provider.rs [hunk 4]
```diff
@@ -738,15 +745,15 @@ fn decode_auto_exchange_error(reason_bytes: Bytes) -> (String, AEReasonError) {
...
         Error::AbiError(revert_reason) => {
-            info!("[Error decoding] ABI error: {:?}", revert_reason);
+            error!("[Error decoding] ABI error: {:?}", revert_reason);
         }
         Error::TransportError(error) => {
             let RpcError::ErrorResp(e) = error else {
-                info!("[Error decoding] Failed to match ErrorResp {:?}", error);
-                return;
+                error!("[Error decoding] Failed to match ErrorResp {:?}", error);
+                return None;
```
Comment (moboware) [inline reya-rust-sdk/src/http_provider.rs:756]: why not return the same error string as in the error! and on the other places too?

File: reya-rust-sdk/src/http_provider.rs [hunk 5]
```diff
@@ -738,15 +745,15 @@ fn decode_auto_exchange_error(reason_bytes: Bytes) -> (String, AEReasonError) {
     }
 }
```
Comment (moboware) [inline reya-rust-sdk/src/http_provider.rs:747]: add a fn description

**Review by moboware — COMMENTED** [review]
add also before the trigger_auto_exchange fn an description

File: reya-rust-sdk/src/http_provider.rs (hunk 4 above)
Comment (ioanaroceanu2000) [reply reya-rust-sdk/src/http_provider.rs:756]: I want to obtain the decoded contract error, if I fail to get it, that will just be an unknown error. Other logs will tell you why it failed to decode

File: reya-rust-sdk/src/http_provider.rs (hunk 5 above)
Comment (ioanaroceanu2000) [reply reya-rust-sdk/src/http_provider.rs:754]: fn descriptions added here [105daad](https://github.com/Reya-Labs/reya-rust-sdks-monorepo/pull/24/commits/105daadbb824bcef6decb5167ba6d80da9d10368)

**Pull request merged**